inc/saiswitch.h

```diff
@@ -3004,6 +3083,26 @@ typedef enum _sai_switch_stat_t
  */
 #define SAI_KEY_HW_PORT_PROFILE_ID_CONFIG_FILE    "SAI_HW_PORT_PROFILE_ID_CONFIG_FILE"
 
+/**
+ * @brief Switch health event callback
+ *
+ * @objects switch_id SAI_OBJECT_TYPE_SWITCH
+ *
+ * @param[in] switch_id Switch Id
+ * @param[in] severity Health event severity
+ * @param[in] timestamp Time and date of receiving the SDK Health event
+ * @param[in] category Category of cause
+ * @param[in] data Data of switch health
+ * @param[in] description Free text with relevant information delivered from SDK event/trap
+ */
+typedef void (*sai_switch_asic_sdk_health_event_notification_fn)(
+        _In_ sai_object_id_t switch_id,
+        _In_ sai_switch_asic_sdk_health_severity_t severity,
+        _In_ sai_timespec_t timestamp,
+        _In_ sai_switch_asic_sdk_health_category_t category,
+        _In_ sai_switch_health_data_t data,
```

Comment:
from definitions so far, this "data" is actually a data type, and only one type of data SAI_HEALTH_DATA_TYPE_GENERAL can be passed.

Reply:
yes currently we don't need to use data for the general type
but we created extendable infrastructure, per Jai's request, so additional events can be added